### PR TITLE
Add gPlayerIsAccelerating, gPlayerBalloonCount and update related code

### PR DESCRIPTION
**Pull request by MaciekBaron**
Starting with a small pull request just to make sure I'm doing things correctly before pushing bigger changes.

The gist of this change is, there is a flag that changes to `1` when a player has acceleration pressed, `0` otherwise.

(Also removed stray white space)

### CODE REVIEW

**Comment by MaciekBaron** [comment]
Realized `spawn_player` calls this and haven't updated that yet, marking as WIP

**Comment by MaciekBaron** [comment]
Alright, I don't think I'm knowledgable enough to understand why I'm getting a mismatch.

When acceleration is pressed, only the 4th byte gets changed - it makes sense to me that the three previous bytes are used for something else, however I only get a match if that is treated as an `s32`.

**Comment by MaciekBaron** [comment]
Changed it to s32 for now so that it matches

File: src/code_800393C0.c [hunk 1]
```diff
@@ -17,7 +17,7 @@ extern Camera *camera4;
 
 //GLOBAL_ASM("asm/non_matchings/code_800393C0/spawn_player.s")
 
-void spawn_player(Player *player, s8 arg1, f32 arg2, f32 arg3, f32 arg4, f32 arg5, u16 arg6, s16 arg7) {
+void spawn_player(Player *player, s8 playerNumber, f32 arg2, f32 arg3, f32 arg4, f32 arg5, u16 arg6, s16 arg7) {
```
Comment (MegaMech) [inline src/code_800393C0.c:20]: I would prefer `playerIndex`

File: include/common_structs.h [hunk 2]
```diff
@@ -391,4 +391,8 @@ typedef struct {
     f32* unk_00[5];
 } struct_D_800E261C;
 
+typedef struct {
+    s32 playerRelatedFlags[8]; // Set to 1 when acceleration is pressed for each player
```
Comment (MegaMech) [inline include/common_structs.h:395]: I'm not really a fan of this name and then having the comment being more specific. I'll take a closer look when I have some more time.

**Review by MegaMech — COMMENTED** [review]
Hi MaciekBaron,

Thanks for your PR!
Just a heads up, that we must not have looked at n64 source code leaks. SDK content from CDs that were sold are generally considered okay though.

File: include/common_structs.h (hunk 2 above)
Comment (MaciekBaron) [reply include/common_structs.h]: I've done some more testing and I don't think those other bytes get ever written to, and they always stay 0. I think the game literally uses the whole 4 bytes for this one flag, which is weird, but there are weirder things in the code. I changed it from a struct to an s32.

**Comment by MaciekBaron** [comment]
> Hi MaciekBaron,
> 
> Thanks for your PR! Just a heads up, that we must not have looked at n64 source code leaks. SDK content from CDs that were sold in auctions are generally considered okay though.

Thanks for the heads-up. I'm just using debugging tools to write the code, definitely not planning to do anything illegal.

I also now added some balloon related code to the same PR.

**Comment by MegaMech** [comment]
This is definitely an s32 in this layout:

```
p1IsAccel, p2IsAccel, p3IsAccel, p4IsAccel
AIp5IsAccel, AIp6IsAccel, AIp7IsAccel, AIp8IsAccel
```

File: src/code_80057C60.c [hunk 3]
```diff
@@ -11100,32 +11100,32 @@ GLOBAL_ASM("asm/non_matchings/code_80057C60/func_8006B87C.s")
...
+    s8 playerIndexTemp;
     u16 *temp_v0;
     u16 temp_t5;
 
-    temp_v1 = (arg1 * 2) + &D_8018D8C0;
-    temp_a3 = *temp_v1;
-    temp_a2 = arg1;
-    if (temp_a3 >= 0) {
-        temp_v0 = (arg1 * 6) + (temp_a3 * 2) + &D_8018D5F0;
+    currentPlayerBalloonCount = (playerIndex * 2) + &gPlayerBalloonCount;
```
Comment (MegaMech) [inline src/code_80057C60.c:11114]: Is this  playerIndex[1]? or playerIndex++ ? or something like that.

File: src/code_80057C60.c [hunk 4]
```diff
@@ -11100,32 +11100,32 @@ GLOBAL_ASM("asm/non_matchings/code_80057C60/func_8006B87C.s")
...
+    playerBalloonCountValue = *currentPlayerBalloonCount;
+    playerIndexTemp = playerIndex;
+    if (playerBalloonCountValue >= 0) {
+        temp_v0 = (playerIndex * 6) + (playerBalloonCountValue * 2) + &D_8018D5F0;
```
Comment (MegaMech) [inline src/code_80057C60.c:11118]: playerIndex[5]?

**Review by MegaMech — APPROVED** [review]

**Comment by MegaMech** [comment]
I was about to merge this but just realized a potential issue with the name.
Where as `gPlayerIsAccelerating` seems pretty accurate. The issue is when you're at max speed you're not really accelerating anymore. However, it equals 1 as long as the throttle button is held down. 


Try something like:
```
gPlayerIsThrottleActive
```

**Comment by MaciekBaron** [comment]
Thanks, that's a good shout!

**Review by MegaMech — APPROVED** [review]

**Pull request merged**